x-pack/plugins/infra/public/components/logging/log_text_stream/log_entry_actions_column.tsx

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useCallback } from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonIcon,
+  EuiPopover,
+  EuiTitle,
+  EuiButtonEmpty,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import { LogEntryColumnContent } from './log_entry_column';
+import { euiStyled } from '../../../../../observability/public';
+
+interface LogEntryActionsColumnProps {
+  isHovered: boolean;
+  isMenuOpen: boolean;
+  openMenu: () => void;
+  closeMenu: () => void;
+  onViewDetails: () => void;
+}
+
+export const LogEntryActionsColumn: React.FC<LogEntryActionsColumnProps> = ({
+  isHovered,
+  isMenuOpen,
+  openMenu,
+  closeMenu,
+  onViewDetails,
+}) => {
+  const handleClickViewDetails = useCallback(() => {
+    closeMenu();
+    onViewDetails();
+  }, [closeMenu, onViewDetails]);
+
+  const label = i18n.translate('xpack.infra.logEntryItemView.logEntryActionsMenuToolTip', {
+    defaultMessage: 'View Details',
+  });
+
+  const button = (
+    <ButtonWrapper>
+      <EuiButtonIcon
+        aria-label={label}
+        color="ghost"
+        iconType="boxesHorizontal"
+        onClick={openMenu}
+      />
+    </ButtonWrapper>
+  );
+
+  return (
+    <ActionsColumnContent>
+      {isHovered ? (
+        <AbsoluteWrapper>
+          <EuiPopover button={button} isOpen={isMenuOpen} closePopover={closeMenu}>
```

Comment:
We recently gained a set of components to build consistent action menus across the observability solutions (https://github.com/elastic/kibana/blob/75aa2076bf9a14342443899fabaffe5c579f7d47/x-pack/plugins/observability/public/components/action_menu.tsx). Would you agree that this looks like a good place to use these?

Reply:
ah, nice! I didn't realise they existed.